src/main.rs

```diff
@@ -128,6 +133,69 @@ fn highlight(q: Json<Query>) -> JsonValue {
     })
 }
 
+#[derive(Deserialize)]
+struct CSSTableQuery {
+    filepath: String,
+    code: String,
+
+    // If set, lines with size greater than line_length_limit will
+    // not be highlighted
+    line_length_limit: Option<usize>,
+}
+
+
+#[post("/css_table", format = "application/json", data = "<q>")]
+fn css_table_index(q: Json<CSSTableQuery>) -> JsonValue {
```

Comment:
I implemented this as a completely separate endpoint because this version returns results in a different format (table vs raw spans). Additionally, it let us remove some deprecated fields from the query.

Reply:
sounds good